src/main/network/packet.rs

```diff
@@ -431,3 +556,54 @@ fn display_udp_bytes(packet: *const c::Packet, mut writer: impl Write) -> std::i
 
     Ok(())
 }
+
+pub fn to_legacy_tcp_flags(flags: tcp::TcpFlags) -> c::ProtocolTCPFlags {
+    let mut new_flags = c::ProtocolTCPFlags_PTCP_NONE;
+
+    for flag in flags.iter() {
+        match flag {
+            tcp::TcpFlags::FIN => new_flags |= c::ProtocolTCPFlags_PTCP_FIN,
+            tcp::TcpFlags::SYN => new_flags |= c::ProtocolTCPFlags_PTCP_SYN,
+            tcp::TcpFlags::RST => new_flags |= c::ProtocolTCPFlags_PTCP_RST,
+            tcp::TcpFlags::PSH => panic!("Unsupported TCP flag: {flag:?}"),
+            tcp::TcpFlags::ACK => new_flags |= c::ProtocolTCPFlags_PTCP_ACK,
+            tcp::TcpFlags::URG => panic!("Unsupported TCP flag: {flag:?}"),
+            tcp::TcpFlags::ECE => panic!("Unsupported TCP flag: {flag:?}"),
+            tcp::TcpFlags::CWR => panic!("Unsupported TCP flag: {flag:?}"),
+            _ => unreachable!(
```

Comment:
Maybe log the value in case this is somehow reached? e.g. maybe just `panic!("Invalid TCP flag: {flag:?}")`

Reply:
Added.